### DISCUSSION TITLE
How to get the value of selected file with useWatch()?

### DISCUSSION THREAD

**Discussion by wick3ds0ul** [react-hook-form/react-hook-form#8556] [Q&A]
Hey guys! 

I am using RHF to select a image file  and render image preview component conditionally if the file was selected or not using the `watch` and `register`.

Here is the setup.

```js



export default function App() {
  const { register, watch } = useForm();
  const watchShowImagePreview = watch("image");

  console.log("Imagewatcher", watchShowImagePreview);
  
  return (
    <div className="App">
      <h1>Select a image file</h1>
      <input type="file" {...register("image")} />
      <br />
      <br />
      {watchShowImagePreview && (
        <img src={watchShowImagePreview[0].name} alt="selected file preview" />
      )}
    </div>
  );
}

```
I am not able to get the value of the file, from the file list.

<img width="1042" alt="Screenshot 2022-06-23 at 8 05 00 PM" src="https://user-images.githubusercontent.com/48680692/175325492-257e722e-9c59-460a-87fa-bd558b902d27.png">

But if I use `useWatch` for a text component it renders the current state of the data.

```js

export default function App() {

  const { register, watch } = useForm();
  const watchShowImagePreview = watch("name");

  return (
    <div className="App">
      <h1>Text Input</h1>
      <input type="text" {...register("name")} />
      <p>{nameWatcher}</p>
     )
   }
   
   ```
   
<img width="659" alt="Screenshot 2022-06-23 at 8 17 43 PM" src="https://user-images.githubusercontent.com/48680692/175327889-df267297-d769-4bed-b749-469863f602e9.png">

   
   Why does it behave differently for files? Please help.
   
   
Reproducable code can be found here in this [sandbox](https://codesandbox.io/embed/modern-glitter-kpfvkx?fontsize=14&hidenavigation=1&theme=dark)

**Comment by bluebill1049** [comment]
File objects have security restruction.

**Reply by wick3ds0ul** [reply]
Is there any way we can have a workaround for this?

I have found a work around for this.

```js
   {watchShowImagePreview && (
        <img src={URL.createObjectURL(watchShowImagePreview[0])} alt="selected file preview" />
      )}
```
